tests/Backend/ExternalBuckets/SnowflakeExternalBucketShareTest.php

```diff
@@ -180,6 +165,132 @@ public function testExternalSchemaAsSharedBucket(): void
+        $this->assertEquals(
+            <<<EXPECTED
+"ID","LASTNAME","GENDER"
+"1","Novák","male"
+
+EXPECTED,
+            $dataPreview,
+        );
+
+        $result = $linkingSnowflakeDb->fetchAll(sprintf(
+            'SELECT * FROM %s.%s.%s',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+        ));
+        $this->assertEquals(
+            [
+                [
+                    'ID' => 1,
+                    'LASTNAME' => 'Novák',
+                    'GENDER' => 'male',
+                ],
+            ],
+            $result,
+        );
+
+        $result2 = $linkingSnowflakeDb->fetchAll(sprintf(
+            'SELECT * FROM %s.%s.%s',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE_2),
+        ));
+        $this->assertEquals(
+            [
+                [
+                    'ID' => 1,
+                    'DESC' => 'important description',
+                ],
+            ],
+            $result2,
+        );
+
+        /** @var \Keboola\Db\Import\Snowflake\Connection $linkingSnowflakeDb */
+        $linkingSnowflakeDb = $linkingBackend->getDb();
+
+        $result3 = $linkingSnowflakeDb->fetchAll(sprintf(
+            'SELECT * FROM %s.%s.%s',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE_2),
+        ));
+        $this->assertEquals(
+            [
+                [
+                    'ID' => 1,
+                    'DESC' => 'important description',
+                ],
+            ],
+            $result3,
+        );
+
+        $db->executeQuery(sprintf(
+            'DROP TABLE %s',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+        ));
+
+        try {
+            $linkingSnowflakeDb->fetchAll(sprintf(
+                'SELECT * FROM %s.%s.%s',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+            ));
+            $this->fail('Should fail.');
+        } catch (Throwable $e) {
+            $this->assertEquals(
+                "odbc_prepare(): SQL error: SQL compilation error:
+Object 'EXT_DB.EXT_SCHEMA.EXT_TABLE' does not exist or not authorized., SQL state S0002 in SQLPrepare",
+                $e->getMessage(),
+            );
+        }
+
+        // REFRESH END
+
         $this->linkingClient->dropBucket($linkedBucketId, ['force' => true]);
```

Comment:
tu to neni vidiel ale potom https://github.com/keboola/storage-api-php-client/blob/888f72451d0935b6c18b7f4980c986440cca8414/tests/Backend/ExternalBuckets/SnowflakeExternalBucketShareTest.php#L296-L307 testujes ze ked unliknes bucket tak nejde cist z `EXTERNAL_TABLE` ale tu si zmazal v tom ex buckete tak len upravit a otestovat ze nejde cist z `EXTERNAL_TABLE_2`(pretoze stale existuje v tom ex buckete) ked unlinknes bucket, aby sme otestovali, ze tam nezostali ziadne granty

Reply:
jasný! to mi nějak včera nedošlo, díky :-)